APIManager.js

```diff
@@ -0,0 +1,78 @@
+class APIManager {
+    constructor() {
+        this.data = {
```

Comment:
You should be able to delete the keys inside this object and it still work. In the code below you are already building the inside of this object. It is just repetative.